src/main/java/net/coderbot/iris/pipeline/HandRenderer.java

```diff
@@ -0,0 +1,130 @@
+package net.coderbot.iris.pipeline;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.PoseStack;
+
+import net.coderbot.iris.layer.GbufferProgram;
+import net.coderbot.iris.mixin.GameRendererAccessor;
+import net.coderbot.iris.uniforms.CapturedRenderingState;
+import net.minecraft.client.Camera;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.GameRenderer;
+import net.minecraft.client.renderer.ItemBlockRenderTypes;
+import net.minecraft.client.renderer.RenderBuffers;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.BlockItem;
+import net.minecraft.world.level.GameType;
+
+public class HandRenderer {
+	public static final HandRenderer INSTANCE = new HandRenderer();
+
+	private static boolean ACTIVE;
+	public static boolean renderingSolid;
+
+	private void setupGlState(GameRenderer gameRenderer, PoseStack poseStack, float tickDelta, Camera camera) {
+        final PoseStack.Pose pose = poseStack.last();
+
+		// We have a inject in getProjectionMatrix to scale the matrix so the hand doesn't clip through blocks.
+		gameRenderer.resetProjectionMatrix(gameRenderer.getProjectionMatrix(camera, tickDelta, false));
+
+        pose.pose().setIdentity();
+        pose.normal().setIdentity();
+
+		if(Minecraft.getInstance().options.bobView) {
+			((GameRendererAccessor)gameRenderer).invokeBobView(poseStack, tickDelta);
+		}
+	}
+
+	private boolean canRender(Camera camera, GameRenderer gameRenderer) {
+		return !(camera.isDetached() 
+			|| !(camera.getEntity() instanceof Player) 
```

Comment:
I don't see vanilla explicitly checking for if the camera entity is a player. Is there a reason for that?

Reply:
As far as I know no, but it's safer.